### PR TITLE
Preserve method comments in JS->ES6 conversion. 

### LINKED ISSUE #16622: Convert To ES6 Class Refactoring Loses JSDocs on Methods
**TypeScript Version:**  2.4.0 / nightly (2.5.0-dev.201xxxxx)

**Code**

```ts
function Wagon(numberOxen) {
    this.numberOxen = numberOxen
}

/**
 * Continue traveling to Oregon
 * @param {number} speed How fast to travel. Should be > 0 and < c
 */
Wagon.prototype.go = function(speed) { }

new Wagon(1)
```

Trigger the `convert to es6 class` refactoring on `Wagon`

**Expected behavior:**

JSDoc on `go` is preserved in the new class

```ts
class Wagon {
    constructor(numberOxen) {
        this.numberOxen = numberOxen;
    }
    /**
     * Continue traveling to Oregon
     * @param {number} speed How fast to travel. Should be > 0 and < c
     */
    go(speed) { }
}

new Wagon(1)
```

**Actual behavior:**
JSdoc is lost:

```ts
class Wagon {
    constructor(numberOxen) {
        this.numberOxen = numberOxen;
    }
    go(speed) { }
}

new Wagon(1)
```

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #16622

cc @rbuckton

### CODE REVIEW

File: src/services/refactors/convertFunctionToEs6Class.ts [hunk 1]
```diff
@@ -183,20 +186,40 @@ namespace ts.refactor {
...
+                        return method;
+                    }
 
-                    default:
+                    default: {
                         // Don't try to declare members in JavaScript files
                         if (isSourceFileJavaScript(sourceFile)) {
                             return;
                         }
                         return createProperty(/*decorators*/ undefined, modifiers, memberDeclaration.name, /*questionToken*/ undefined,
```
Comment (rbuckton) [inline src/services/refactors/convertFunctionToEs6Class.ts]: Should we copy comments for properties as well?

**Comment by RyanCavanaugh** [comment]
@rbuckton review latest rev?

File: src/services/refactors/convertFunctionToEs6Class.ts [hunk 2]
```diff
@@ -183,20 +186,42 @@ namespace ts.refactor {
                             const expression = arrowFunctionBody as Expression;
                             bodyBlock = createBlock([createReturn(expression)]);
                         }
-                        return createMethod(/*decorators*/ undefined, modifiers, /*asteriskToken*/ undefined, memberDeclaration.name, /*questionToken*/ undefined,
+                        const method = createMethod(/*decorators*/ undefined, modifiers, /*asteriskToken*/ undefined, memberDeclaration.name, /*questionToken*/ undefined,
```
Comment (rbuckton) [inline src/services/refactors/convertFunctionToEs6Class.ts]: Can you add a test for this?

File: src/services/refactors/convertFunctionToEs6Class.ts [hunk 3]
```diff
@@ -183,20 +186,42 @@ namespace ts.refactor {
...
+                    }
 
-                    default:
+                    default: {
                         // Don't try to declare members in JavaScript files
                         if (isSourceFileJavaScript(sourceFile)) {
                             return;
                         }
-                        return createProperty(/*decorators*/ undefined, modifiers, memberDeclaration.name, /*questionToken*/ undefined,
+                        const prop = createProperty(/*decorators*/ undefined, modifiers, memberDeclaration.name, /*questionToken*/ undefined,
```
Comment (rbuckton) [inline src/services/refactors/convertFunctionToEs6Class.ts]: Can you add a test for this?

Comment (RyanCavanaugh) [inline src/services/refactors/convertFunctionToEs6Class.ts]: https://github.com/Microsoft/TypeScript/pull/16697/files#diff-5cfc913717c53d115c9368927495ac25R6

Comment (rbuckton) [inline src/services/refactors/convertFunctionToEs6Class.ts]: That only hits the code path on [L167](https://github.com/Microsoft/TypeScript/pull/16697/files#diff-77ef55c503714939dff4c55944ea08ffL167). It doesn't cover the cases on L186 and L189.

Comment (RyanCavanaugh) [inline src/services/refactors/convertFunctionToEs6Class.ts]: This is actually currently unreachable since this refactor doesn't light up in non-JS files

Comment (rbuckton) [inline src/services/refactors/convertFunctionToEs6Class.ts]: Fair point regarding the property declaration, but arrow functions are still fair game.

File: src/services/refactors/convertFunctionToEs6Class.ts (hunk 2 above)
Comment (rbuckton) [inline src/services/refactors/convertFunctionToEs6Class.ts]: Not related to comments, but do we preserve the binding to the `this` of the outer scope if the arrow contains a `this`? What happens if we refactor:

```js
function C() {}
C.prototype.x = () => this;
```

**Comment by RyanCavanaugh** [comment]
We don't preserve arrow functions' `this` during refactors. I think the assumption is that people are mostly writing things like `fn.prototype.bar = () => 10;` (obviously we could syntactically distinguish the two but it seems like an odd thing to write). Since arrow functions and classes landed in the same ES version I don't expect to see much of this.

**Comment by RyanCavanaugh** [comment]
Thanks!

**Comment by rbuckton** [comment]
Can you merge this into release-2.4?

**Pull request merged**